docs/containers-registries.conf.5.md

```diff
@@ -109,6 +109,18 @@ internet without having to change `Dockerfile`s, or to add redundancy).
 *Note*: Redirection and mirrors are currently processed only when reading images, not when pushing
 to a registry; that may change in the future.
 
+#### Normalization of docker.io references
+
+The Docker Hub `docker.io` is handled in a special way, because every push and
+pull operation gets internally normalized with `/library` if no other specific
+namespace is defined (for example on `docker.io/username/image`).
+
+This means that a pull of `docker.io/alpine` will be internally translated to
+`docker.io/library/alpine`. A pull of `docker.io/user/alpine` will not be
+rewritten because this is already the correct remote path.
```

Comment:
This version may have overshot the “let’s not give the users a misleading example” direction, in particular I’m a bit worried that the users who read this will not connect this statement to its impact on the mirror/redirection features.

Maybe add a paragraph here vaguely to the effect of
> Therefore, to remap/mirror the `docker.io` images in the implied `/library` namespace (or that whole namespace), the `prefix` and `location` fields in this configuration file must explicitly include that `/library` namespace (e.g. `prefix = "docker.io/library/alpine"`, not `prefix = "docker.io/alpine"`; the latter would match the `docker.io/alpine/*` repositories but not the `docker.io/[library/]alpine` image).

OTOH this one is probably over-explaining yet again, please edit as desired.

Reply:
I adapted your paragraph and changed it a bit.